File: docs/snippets/all/tutorials/getting_started_log.py
import math

import rerun as rr

with rr.RecordingStream("rerun_example_getting_started", recording_id="run-1", send_properties=False) as rec:
    rec.save("run-1.rrd")
    for t in range(10):
        rec.set_time("t", duration=t)
        rec.log("/arm/shoulder", rr.Scalars(math.sin(t * 0.5)))
        rec.log("/arm/elbow", rr.Scalars(math.cos(t * 0.5)))

File: docs/snippets/all/tutorials/getting_started_log.cpp
#include <rerun.hpp>

#include <cmath>

int main(int argc, char* argv[]) {
    const auto rec =
        rerun::RecordingStream("rerun_example_getting_started", "run-1");
    rec.save("run-1.rrd").exit_on_failure();

    for (int t = 0; t < 10; ++t) {
        const auto tf = static_cast<double>(t);
        rec.set_time_duration_secs("t", tf);
        rec.log("/arm/shoulder", rerun::Scalars(std::sin(tf * 0.5)));
        rec.log("/arm/elbow", rerun::Scalars(std::cos(tf * 0.5)));
    }
}
